library/src/main/java/com/novoda/downloadmanager/DownloadsNetworkRecoveryCreator.java

```diff
@@ -2,23 +2,27 @@
 
 import android.content.Context;
 
-class DownloadsNetworkRecoveryCreator {
+final class DownloadsNetworkRecoveryCreator {
 
-    private static DownloadsNetworkRecovery INSTANCE;
+    private DownloadsNetworkRecoveryCreator() {
+        // Uses static factory methods.
+    }
+
+    private static DownloadsNetworkRecovery singleInstance;
```

Comment:
shouldn't the constant be upper case?

Reply:
It's not a constant. It's only a constant when it is `static` and `final`. This is still mutable, just at the class level.